lib/models/key.coffee

```diff
@@ -14,126 +14,135 @@ See the License for the specific language governing permissions and
-# @public
-# @function
-# @memberof resin.models.key
-#
-# @param {String} title - key title
-# @param {String} key - the public ssh key
-#
-# @fulfil {Object} - ssh key
-# @returns {Promise}
-#
-# @example
-# resin.models.key.create('Main', 'ssh-rsa AAAAB....').then(function(key) {
-# 	console.log(key);
-# });
-#
-# @example
-# resin.models.key.create('Main', 'ssh-rsa AAAAB....', function(error, key) {
-# 	if (error) throw error;
-# 	console.log(key);
-# });
-###
-exports.create = (title, key, callback) ->
+	###*
+	# @summary Get all ssh keys
+	# @name getAll
+	# @public
+	# @function
+	# @memberof resin.models.key
+	#
+	# @fulfil {Object[]} - ssh keys
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.key.getAll().then(function(keys) {
+	# 	console.log(keys);
+	# });
+	#
+	# @example
+	# resin.models.key.getAll(function(error, keys) {
+	# 	if (error) throw error;
+	# 	console.log(keys);
+	# });
+	###
+	exports.getAll = (callback) ->
+		return pine.get
+			resource: 'user__has__public_key'
+		.asCallback(callback)
 
-	# Avoid ugly whitespaces
-	key = key.trim()
+	###*
+	# @summary Get a single ssh key
+	# @name get
+	# @public
+	# @function
+	# @memberof resin.models.key
+	#
+	# @param {(String|Number)} id - key id
+	# @fulfil {Object} - ssh key
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.key.get(51).then(function(key) {
+	# 	console.log(key);
+	# });
+	#
+	# @example
+	# resin.models.key.get(51, function(error, key) {
+	# 	if (error) throw error;
+	# 	console.log(key);
+	# });
+	###
+	exports.get = (id, callback) ->
+		return pine.get
+			resource: 'user__has__public_key'
+			id: id
+		.tap (key) ->
+			if isEmpty(key)
+				throw new errors.ResinKeyNotFound(id)
+		.asCallback(callback)
 
-	auth.getUserId().then (userId) ->
-		return pine.post
+	###*
+	# @summary Remove ssh key
+	# @name remove
+	# @public
+	# @function
+	# @memberof resin.models.key
+	#
+	# @param {(String|Number)} id - key id
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.key.remove(51);
+	#
+	# @example
+	# resin.models.key.remove(51, function(error) {
+	# 	if (error) throw error;
+	# });
+	###
+	exports.remove = (id, callback) ->
+		return pine.delete
 			resource: 'user__has__public_key'
-			body:
-				title: title
-				public_key: key
-				user: userId
-	.nodeify(callback)
+			id: id
+		.asCallback(callback)
+
+	###*
+	# @summary Create a ssh key
+	# @name create
+	# @public
+	# @function
+	# @memberof resin.models.key
+	#
+	# @param {String} title - key title
+	# @param {String} key - the public ssh key
+	#
+	# @fulfil {Object} - ssh key
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.key.create('Main', 'ssh-rsa AAAAB....').then(function(key) {
+	# 	console.log(key);
+	# });
+	#
+	# @example
+	# resin.models.key.create('Main', 'ssh-rsa AAAAB....', function(error, key) {
+	# 	if (error) throw error;
+	# 	console.log(key);
+	# });
+	###
+	exports.create = (title, key, callback) ->
+
+		# Avoid ugly whitespaces
+		key = key.trim()
```

Comment:
If `key` isn't a string this will throw, breaking the expectation that a promise will always be returned